lib/bloc/podcast_history/model.dart

```diff
@@ -0,0 +1,239 @@
+      default:
+        return PodcastHistoryTimeRangeDaily();
+    }
+  }
+
+  String toJson() {
+    return json.encode({
+      "type": _type(),
+      "startDate": startDate.millisecondsSinceEpoch,
+      "endDate": endDate.millisecondsSinceEpoch,
+    });
+  }
+}
+
+class PodcastHistoryTimeRangeDaily extends PodcastHistoryTimeRange {
+  PodcastHistoryTimeRangeDaily()
+      : super._(
+          DateTime(
+            DateTime.now().year,
+            DateTime.now().month,
+            DateTime.now().day,
+            0,
+            0,
+            0,
+          ),
+          DateTime(
+            DateTime.now().year,
+            DateTime.now().month,
+            DateTime.now().day,
+            23,
+            59,
+            59,
+            999,
+          ),
+        );
+
+  @override
+  int _type() => _podcastHistoryTimeRangeDaily;
+}
+
+class PodcastHistoryTimeRangeWeekly extends PodcastHistoryTimeRange {
+  PodcastHistoryTimeRangeWeekly()
+      : super._(
+          DateTime(
+            DateTime.now().year,
+            DateTime.now().month,
+            DateTime.now().subtract(Duration(days: DateTime.now().weekday)).day,
+            0,
+            0,
+            0,
+          ),
+          DateTime.now(),
+        );
+
+  @override
+  int _type() => _podcastHistoryTimeRangeWeekly;
+}
+
+class PodcastHistoryTimeRangeMonthly extends PodcastHistoryTimeRange {
+  PodcastHistoryTimeRangeMonthly()
+      : super._(
+          DateTime(
+            DateTime.now().year,
+            DateTime.now().month,
+            1,
+            0,
+            0,
+            0,
+          ),
+          DateTime.now(),
+        );
+
+  @override
+  int _type() => _podcastHistoryTimeRangeMonthly;
+}
+
+class PodcastHistoryTimeRangeYearly extends PodcastHistoryTimeRange {
+  PodcastHistoryTimeRangeYearly()
+      : super._(
+          DateTime(
+            DateTime.now().year - 1,
+            DateTime.now().month,
+            1,
+            0,
+            0,
+            0,
+          ),
+          DateTime.now(),
+        );
+
+  @override
+  int _type() => _podcastHistoryTimeRangeYearly;
+}
+
+class PodcastHistoryTimeRangeAllTime extends PodcastHistoryTimeRange {
+  PodcastHistoryTimeRangeAllTime()
+      : super._(
+          DateTime(
+            _allTimeRangeYear,
+            DateTime.now().month,
+            1,
+            0,
+            0,
+            0,
+          ),
+          DateTime.now(),
+        );
+
+  @override
+  int _type() => _podcastHistoryTimeRangeAllTime;
+}
+
+class PodcastHistoryRecord {
+  final int totalSatsStreamedSum;
+  final int totalBoostagramSentSum;
+  final double totalDurationInMinsSum;
+  List<PodcastHistoryModel> podcastHistoryList;
+
+  PodcastHistoryRecord(
+      {this.totalSatsStreamedSum,
+      this.totalBoostagramSentSum,
+      this.totalDurationInMinsSum,
+      this.podcastHistoryList});
+}
+
+abstract class PodcastHistorySortOptions {
```

Comment:
It seems that we don't need this class as we can use directly the enum

Reply:
@roeierez  Yes. Also, I think it's better if we do not store the selected sort options in `shared_preferences` this looks confusing and other apps don't do it.

Reply:
@utkarshshendge  so why did you implement it?

Reply:
Done